File: coinsaver-api/src/main/java/com/coinsaver/api/dtos/response/TransactionResponseDto.java
package com.coinsaver.api.dtos.response;

import com.coinsaver.core.enums.StatusType;
import com.coinsaver.core.enums.TransactionCategoryType;
import com.coinsaver.core.enums.TransactionType;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
public class TransactionResponseDto {

    private Long transactionId;
    private Long installmentTransactionId;
    private Long fixTransactionId;
    private BigDecimal amount;
    private LocalDate payDay;
    private String description;
    private StatusType status;
    private TransactionCategoryType category;
    private Boolean fixedExpense;
    private Integer repeat_;
    private TransactionType transactionType;
    private DivisionResponseDto division;
}

File: coinsaver-front/src/app/dtos/transactions/response/transaction.response.dto.ts
import { TransactionCategoryTypeEnum } from 'src/app/enums/transaction-category-type.enum';
import { TransactionTypeEnum } from 'src/app/enums/transaction-type.enum';
import { StatusTypeEnum } from '../../../enums/status-type.enum';

export class TransactionResponseDto {
  constructor(
    public transactionId: number,
    public installmentTransactionId: number,
    public fixTransactionId: number,

    public amount: number,

    public payDay: string | Date,

    public description: string,

    public status: StatusTypeEnum,
    public category: TransactionCategoryTypeEnum,

    public fixedExpense: boolean,

    public transactionType: TransactionTypeEnum,

    public repeat: number,
  ) {}
}
